### PR TITLE
Expression type info in validation result

### PR DESCRIPTION
**Pull request by nadberezny**
1. Add expression TypingResult to ValidationResult
2. Display inferred expression type on modal open

### CODE REVIEW

File: ui/restmodel/src/main/scala/pl/touk/nussknacker/restmodel/validation/ValidationResults.scala [hunk 1]
```diff
@@ -54,7 +54,9 @@ object ValidationResults {
   }
 
   object NodeTypingData {
-    implicit val typingInfoEncoder: Encoder[ExpressionTypingInfo] = Encoder.instance(_ => Json.Null)
+    implicit val typingInfoEncoder: Encoder[ExpressionTypingInfo] = Encoder.instance { expressionTypingInfo =>
+      TypeEncoders.typingResultEncoder.apply(expressionTypingInfo.typingResult)
```
Comment (nadberezny) [inline ui/restmodel/src/main/scala/pl/touk/nussknacker/restmodel/validation/ValidationResults.scala:58]: Is this ok?

File: engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/sql/SqlExpressionParser.scala [hunk 2]
```diff
@@ -116,7 +116,10 @@ class SqlExpression(private[sql] val columnModels: Map[String, ColumnModel],
 }
 
 // empty for now
-case object SqlExpressionTypingInfo extends ExpressionTypingInfo
+case object SqlExpressionTypingInfo extends ExpressionTypingInfo {
+
+  override def typingResult: TypingResult = typing.Unknown
```
Comment (mproch) [inline engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/sql/SqlExpressionParser.scala:121]: I think we can safely change it to case class with typingResult - we use it in place where it's already determined

File: engine/interpreter/src/test/scala/pl/touk/nussknacker/engine/InterpreterSpec.scala [hunk 3]
```diff
@@ -813,6 +813,9 @@ object InterpreterSpec {
     = Valid(LiteralExpression(original))
   }
 
-  case object LiteralExpressionTypingInfo extends ExpressionTypingInfo
+  case object LiteralExpressionTypingInfo extends ExpressionTypingInfo {
+
+    override def typingResult: TypingResult = typing.Unknown
```
Comment (mproch) [inline engine/interpreter/src/test/scala/pl/touk/nussknacker/engine/InterpreterSpec.scala:818]: same as above

File: ui/restmodel/src/main/scala/pl/touk/nussknacker/restmodel/validation/ValidationResults.scala (hunk 1 above)
Comment (mproch) [reply ui/restmodel/src/main/scala/pl/touk/nussknacker/restmodel/validation/ValidationResults.scala:58]: It should work, but I think it's easier to write TypeEncoders.typingResultEncoder.contramap[ExpressionTypingInfo](_.typingResult) :)

File: ui/server/src/main/scala/pl/touk/nussknacker/ui/uiresolving/UIProcessResolving.scala [hunk 4]
```diff
@@ -43,7 +43,8 @@ class UIProcessResolving(validation: ProcessValidation, substitutorByProcessingT
       .getOrElse(canonical)
     val displayable = ProcessConverter.toDisplayable(substituted, processingType, businessView)
     val uiValidations = validation.uiValidation(displayable)
-    new ValidatedDisplayableProcess(displayable, uiValidations.add(validationResult).withClearedTypingInfo)
```
Comment (mproch) [inline ui/server/src/main/scala/pl/touk/nussknacker/ui/uiresolving/UIProcessResolving.scala:46]: Shouldn't we remove all withClearedTypingInfo for UI to work in all cases?

File: engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/sql/SqlExpressionParser.scala (hunk 2 above)
Comment (nadberezny) [reply engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/sql/SqlExpressionParser.scala:121]: 👍

File: engine/interpreter/src/test/scala/pl/touk/nussknacker/engine/InterpreterSpec.scala (hunk 3 above)
Comment (nadberezny) [reply engine/interpreter/src/test/scala/pl/touk/nussknacker/engine/InterpreterSpec.scala:818]: 👍

File: ui/restmodel/src/main/scala/pl/touk/nussknacker/restmodel/validation/ValidationResults.scala (hunk 1 above)
Comment (nadberezny) [reply ui/restmodel/src/main/scala/pl/touk/nussknacker/restmodel/validation/ValidationResults.scala:58]: (ﾉ◕ヮ◕)ﾉ*:･ﾟ✧
👍

File: ui/server/src/main/scala/pl/touk/nussknacker/ui/uiresolving/UIProcessResolving.scala (hunk 4 above)
Comment (nadberezny) [reply ui/server/src/main/scala/pl/touk/nussknacker/ui/uiresolving/UIProcessResolving.scala:46]: done:
https://github.com/TouK/nussknacker/pull/1222/commits/ea7704364eca5bb38d774d8176ddb11693c91d7a

**Review by mproch — APPROVED** [review]
:+1:

File: ui/server/src/test/scala/pl/touk/nussknacker/ui/process/marshall/ProcessConverterSpec.scala [hunk 5]
```diff
@@ -123,7 +123,14 @@ class ProcessConverterSpec extends FunSuite with Matchers with TableDrivenProper
       )
     )
 
-    displayableCanonical(process.toDisplayable) shouldBe process
+    val displayableProcess = displayableCanonical(process.toDisplayable)
+    // because I'm lazy
```
Comment (arkadius) [inline ui/server/src/test/scala/pl/touk/nussknacker/ui/process/marshall/ProcessConverterSpec.scala:127]: :laughing:

**Review by arkadius — APPROVED** [review]

**Pull request merged**